### ISSUE TITLE
periodic boundary conditions for arbitrary mesh files?

### ISSUE THREAD

**Issue by stevengj** [gridap/Gridap.jl#412]
We saw in #266 and #318 that you support periodic boundary conditions for `CartesianDiscreteModel`.   What would be needed to support them for arbitrary (periodic) meshes?

In particular, we are looking at periodic triangular meshes in a rectangular domain, generated by Gmsh, and would like a way to tell Gridap's matrix assembly to treat one of the directions as periodic.

**Comment by stevengj** [comment]
@fverdugo, if you have a chance, it would be great to get a few hints from you on this.

**Comment by fverdugo** [comment]
It should be possible. One has to understand how Gmsh encodes information about periodic boundaries and then use it in Gridap.  Essentially, one needs to pair the vertices at the two ends of a periodic boundary from the info provided by gmsh. The rest should be easy.

@jesusbonilla and @AndiMD have recently worked with GridapGmsh and perhaps they are also interested in this feature.

**Comment by jesusbonilla** [comment]
@stevengj I started to work on this point in a [fork of GridapGmsh](https://github.com/jesusbonilla/GridapGmsh.jl/tree/import_mesh_with_periodic_bc) ([these are the main modifications](https://github.com/jesusbonilla/GridapGmsh.jl/blob/import_mesh_with_periodic_bc/src/GmshDiscreteModels.jl#L7-L76)). However, the approach that I have followed so far might be quite basic/preliminary. It is still not working properly. I am able to construct the `gird_topology`, but some development is required to label faces properly. I am also assuming only one periodic direction (extension to multiple should be straightforward though) and I assume that the mesh is periodic, i.e., I am not checking that it is actually periodic (something that IMO should be done). I would like to have this working at some point, but unfortunately I will not be able to devote much time to this in the next weeks :(
